pyfr/solvers/base/system.py

```diff
@@ -62,13 +62,18 @@ def __init__(self, backend, rallocs, mesh, initsoln, nregs, cfg):
         self._bc_inters = self._load_bc_inters(rallocs, mesh, elemap)
         backend.commit()
 
+        self._has_src_macro = False
```

Comment:
Make this a property.

Reply:
Slightly unsure as to how best to do this. As I wrote it self._has_src_macro has to be set in commit() on L74 below, after kernels are generated but before ele_map is deleted. So do I need to keep the private self._has_src_macro about, and then have a property has_src_macro that returns _has_src_macro? And/or do I need to define a setter method for the property?

Reply:
From the variables which we already have we can always determine if we have an src macro or not.  Thus we can make this a property rather than its own variable.